### PR TITLE
Allow user provide more detail to YCSB

### PR DESCRIPTION
**Pull request by jtaleric**
This allows the user to modify the YCSB workload a bit.

### CODE REVIEW

File: roles/load-ycsb/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        template:
+          metadata:
+            name: '{{ meta.name }}-ycsb-data-load'
+            namespace: '{{ operator_namespace }}'
+            labels:
+             name: 'ycsb-load'
+          spec:
+            containers:
+            - name: ycsb
+              image: "quay.io/jtaleric/ycsb-server:testing"
```
Comment (dry923) [inline roles/load-ycsb/tasks/main.yml]: Should the image location be in quay/benchmark-operator?

File: roles/ycsb-bench/tasks/main.yml [hunk 2]
```diff
@@ -66,11 +25,33 @@
           spec:
             containers:
             - name: ycsb
-              image: "quay.io/cloud-bulldozer/ycsb-server:latest"
+              image: "quay.io/jtaleric/ycsb-server:testing"
```
Comment (dry923) [inline roles/ycsb-bench/tasks/main.yml]: Same as above w/ re the image location.

**Comment by dry923** [comment]
This is failing in kni and minishift because it is trying to write to /data/db which would be a privileged location. Changing the mongod default db path to tmp fixes this in my test on kni:

 args:  ["-c", "mkdir -p /tmp/data/db; mongod --smallfiles --bind_ip 0.0.0.0 --dbpath /tmp/data/db"]

File: roles/load-ycsb/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+user: ""
+elasticsearch:
+        server: ""
+        port: ""
```
Comment (dustinblack) [inline roles/load-ycsb/defaults/main.yml]: I've been struggling to get defaults to work the way I expect -- applying for missing, empty, or blank values. This method only takes care of missing, but empty or blank will be passed on as blank. Not a blocker problem at all, but something that could reduce user error if we can fix it.

File: roles/load-ycsb/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+              args: ["-c", "/opt/snafu/ycsb-wrapper/ycsb-wrapper.py -r 1 -l -d {{ ycsb.driver }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
+              volumeMounts:
+                - name: config-volume
+                  mountPath: "/tmp/ycsb"
+            restartPolicy: Never
+            volumes:
+            - name: config-volume
+              configMap:
+                name: ycsb-test
+        backoffLimit: 4
```
Comment (dustinblack) [inline roles/load-ycsb/tasks/main.yml]: Do you want to pull this job out into a template to clean up the task file a bit?

File: roles/load-ycsb/templates/configmap.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    recordcount={{ycsb.recordcount}}
+    operationcount={{ycsb.operationcount}}
+    workload=com.yahoo.ycsb.workloads.CoreWorkload
+    readallfields=true
+    readproportion=0.5
+    updateproportion=0
+    scanproportion=0
+    insertproportion=0
+    readmodifywriteproportion=0.5
+    requestdistribution=zipfian
```
Comment (dustinblack) [inline roles/load-ycsb/templates/configmap.yml.j2]: You might take a look at the recent work I've done for fio that allows for overriding config defaults via the CR file. Something similar here might provide even more flexibility for the power user.

File: roles/ycsb-bench/tasks/main.yml [hunk 6]
```diff
@@ -67,10 +26,32 @@
...
-              args: ["-c", "/ycsb/bin/ycsb run {{ ycsb.driver }} -s -P /ycsb/workloads/{{ ycsb_workload }} {{ ycsb.options_load }}"]
+              args: ["-c", "/opt/snafu/ycsb-wrapper/ycsb-wrapper.py -r 1 -d {{ ycsb.driver }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
+              volumeMounts:
+                - name: config-volume
+                  mountPath: "/tmp/ycsb"
             restartPolicy: Never
+            volumes:
+              - name: config-volume
+                configMap:
+                  name: ycsb-test
```
Comment (dustinblack) [inline roles/ycsb-bench/tasks/main.yml]: Similar comment about pulling this out to a template file.

File: roles/load-ycsb/defaults/main.yml (hunk 3 above)
Comment (jtaleric) [reply roles/load-ycsb/defaults/main.yml]: Ah, so this is due to me moving to passing the env vars via the container yaml. If the user does not pass the `elasticsearch` var ansible will throw up. I had to create these defaults... However, the SNAFU code should ignore empty value server/port.

File: roles/load-ycsb/tasks/main.yml (hunk 4 above)
Comment (jtaleric) [reply roles/load-ycsb/tasks/main.yml]: I thought about this. It might not be a bad idea.

File: roles/load-ycsb/templates/configmap.yml.j2 (hunk 5 above)
Comment (jtaleric) [reply roles/load-ycsb/templates/configmap.yml.j2]: Correct, I just got a email from Elko and he mentioned that we should look at changing this to uniform. I should template this out too...

**Pull request merged**